src/util/tools.cc

```diff
@@ -229,13 +229,9 @@ fs::path find_in_path(const fs::path& exec)
 
     std::error_code ec;
     auto pathAr = split_string(PATH, ':');
-    for (auto& path : pathAr) {
-        fs::path check = fs::path(path) / exec;
-        if (isRegularFile(check, ec))
-            return check;
-    }
+    auto it = std::find_if(pathAr.begin(), pathAr.end(), [&](const auto& path) { fs::path check = fs::path(path) / exec; return isRegularFile(check, ec); });
```

Comment:
This is not an equivalent transformation: before the full path was returned, now only the base path.

Honestly I'm not convinced that converting this loop worth the effort.

Reply:
Good catch. Reverted.